src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/Extensions/ContextQuery/SyntaxTreeExtensions.vb

```diff
@@ -24,7 +24,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Extensions.ContextQuery
                 End If
 
                 ' If this token is to our left, return it
-                If Not token.IsKind(SyntaxKind.EmptyToken) AndAlso token.Span.End < position Then
+                If Not token.IsKind(SyntaxKind.EmptyToken) AndAlso token.Span.End <= position Then
```

Comment:
This now introduces many other test failures and some assertion failures..
This should be callers error right?

Reply:
One possible solution here is to add an optional parameter `canReturnTokenEndingInPosition` and this parameter will determine whether `<` or `<=` should be used. While this could be the easiest solution, I feel it's a bit ugly, and unclear. It's more of a workaround rather than a "real" solution/fix.

Reply:
or just check for exactly this case here.

Reply:
> or just check for exactly this case here.

Not sure what you mean by that?

Reply:
chekc for the `if(` case here instead of the generic approach you're taking.

Reply:
@CyrusNajmabadi I think this might not guarantee that other language constructs will work as expected. At the same time, this won't guarantee that other callers expecting the `<` case will work when the token is IfKeyword.